specs/networking/libp2p-standardization.md

```diff
@@ -183,42 +197,53 @@ Each RPC message is segregated into it's own libp2p protocol id, which is a stri
 ```
 
 With:
-* **ProtocolPrefix** -- the RPC messages are grouped into families identified by a shared LibP2P protocol name prefix. A conforming implementation is expected to support either all messages appearing within a family or none of them. In this case, we use `/eth/serenity/rpc`.
-* **MessageName** -- each RPC request is identified by a name consisting of English letters, digits and underscores (_).
-* **SchemaVersion** -- a semantic version consisting of one or more numbers separated by dots (.). Each schema is versioned to facilitate backward and forward-compatibility when possible.
-* **Encoding** -- while the schema defines the data types in more abstract terms, the encoding describes a specific representation of bytes that will be transmitted over the wire. See the [Encodings](#encodings) section, for further details.
-
-This protocol segregation allows libp2p `multistream-select` to handle the RPC-type, version and encoding negotiation before establishing the underlying substreams.
+* **ProtocolPrefix** -- the RPC messages are grouped into families identified
+  by a shared LibP2P protocol name prefix. A conforming implementation is
+  expected to support either all messages appearing within a family or none of
+  them. In this case, we use `/eth/serenity/rpc`.
```

Comment:
I have no strong objections to using this prefix, but I'll share my reasoning for why I used `/ETH/BeaconChain/` in my own proposal. 

The current RPC procs describe everything necessary to sync and propagate changes to the beacon chain. As soon as we add shards to the mix, we'll need new RPC procs for syncing the state of shards. Custom execution environments may also need their own specialized protocols for communicating with the relayers in the network.

"Serenity" is a term that encompasses all of these future developments and the wording in the spec suggests that a node should support either all of the protocols or none of them, but I see some potential for more specialized software that syncs only with the beacon chain for example. 

Thus, I tried to come up with the most unambiguous prefix name that is unlikely to clash with anything else in the future. "/ETH/BeaconChain/" seemed to fit that bill. The choice of PascalCase was another silly concern trying to win 1 byte in the multiselect handshakes :)

Reply:
I am also not strongly opinionated on this, but would opt for whatever we go for, to be somewhat consistent with all protocols. 
Hopefully there will be more input by someone and I'm happy to go with whatever is the consensus.

Reply:
If we are to go with `/eth/beacon_chain`, that would imply what other protocols...

* `/eth/shard` -- specific requests to shard chains, likely most calls would have 1st param as `shard_id`
* `/eth/network` -- akin to `net_` methods in eth1 rpc

Any others?

Another question is should the initial component of the prefix be `/eth2` -- potentially allows for better segregation if components of eth1 end up using the libp2p for any part of the stack

Reply:
> Any others?

I guess everything phase 2 related -- propagating transactions, syncing state, requesting light client proofs, etc., and everything replicated for the different execution environments.

> Another question is should the initial component of the prefix be /eth2 -- potentially allows for better segregation if components of eth1 end up using the libp2p for any part of the stack

Originally, I preferred `eth2`, mostly because everything else is called `eth2` as well. But one could think of cases in which `eth1` and `eth2` clients would want to speak with each other and then having a different prefix might be awkward (thinking about this idea of storing raw block data in Swarm or another sort of persistence layer). But even then I guess we could have an entirely different prefix for those cases?

Reply:
To make this discussion more concrete, there is only really two protocol Id's we currently need to specify.  Gossipsub and the RPC. The topics within gossipsub can be set independently and without a prefix. 
Is the consensus:
`gossipsub`: `/eth2/beacon_chain/gossipsub/<version>/<encoding>`
`rpc`: `/eth2/beacon_chain/rpc/<version>/<encoding>` 
?